Added functionallity to handle marking of capital letters in the same way as hyphenation.

kalaspuffar opened the pull request:
Hi @bertfrees and @PaulRambags

After reviewing the code for how the hyphenation code is implemented, I realized that it was done with an attribute, and I see no reason why capital letter marks could not be handled the same way.

This pull request adds the functionality of a new attribute for text processing. mark-capital-letters could be used to inform the formatter if the text should have capital letter marks.

From OBFL Specification https://github.com/mtmse/obfl/pull/28

Please review and comment.

Best regards
Daniel

PaulRambags reviewed (changes requested):
I couldn't find a description of this functionality in the OBFL specification. Did I overlook it?
If not, please add a description of this marking of capital letters in the OBFL specification.

kalaspuffar commented:
Hi @PaulRambags 

I've added the OBFL PR to the description. Still under discussion. I just moved the PR to the new repository.

Best regards
Daniel

kalaspuffar commented on src/org/daisy/dotify/api/formatter/TextProperties.java:
```diff
@@ -96,11 +109,21 @@ public boolean isHyphenating() {
+     * Returns true if the mark capital letters property is true, false otherwise
+     * @return returns true if the mark capital letters property is true
+     */
+    public boolean shouldMarkCapitalLetters() {
```
Hi @PaulRambags

When it comes to the naming of this function, I use the most reasonable verb. If you want another example of this use, you have it with the hyphenate function: https://github.com/mtmse/dotify.library/blob/0e7b130d72a2355b995a7a0c97abe821c065dce0/src/org/daisy/dotify/translator/DefaultBrailleFilter.java#L128

In the specific case, the boolean doesn't tell you if the current object is a mark capital letter. "is" as a keyword is really great for telling about a state of an object. But in this case, it tells us if we should run a specific piece of code.

The style guide of Oracle only tells you to use a relevant verb for your function names:
https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

This can be a part that we could debate what verbs are allowed in this codebase, and then I will change it. But in that change, I will go through and change all other usages of the same keywords, so we get a cohesive use.

Best regards
Daniel
